### PR TITLE
HPCC-32146 Fix logic waiting for workunit complete to spot hard fail

### PR DESCRIPTION
**Pull request by jakesmith**
The workunit k8s launch code, waits on the state of the workunit after it's lauched. Once it responds, which includes if it disappears and the queuing agent (thoragent) marks it as fails, it scans the workuinit for exceptions and throws one if it exists. However, if there were some exceptions, but no 'thormasterexception' reply exception, then the logic skipped the follow on test to see if the workunit had been marked failed, and continued.

This results in either jobs appearing to run indefinitely, or spuriously marking themselves as complete

NB: reproduced by deleting the thormanager pod abruptly (--force=true)

There also some changes her which wrap the deleteResource commands in case they throw exception, because we do not want those exceptions to cause job failures in themselves, or mask the original exception. Log as operator warnings instead.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
Perhaps it is not related, but on Rocky-8 there is a strange error, related to this PR
```
   434. Test: javaembed_ex5.ecl
    eclccserver: (0,0): Error C9999: eclcc killed - likely to be out of memory - see compile log for details
    1 error(s), 0 warning(s)
```
@ghalliday ?

**Comment by jakesmith** [comment]
> Perhaps it is not related, but on Rocky-8 there is a strange error on thor, related to this PR
> 
> ```
>    434. Test: javaembed_ex5.ecl
>     eclccserver: (0,0): Error C9999: eclcc killed - likely to be out of memory - see compile log for details
>     1 error(s), 0 warning(s)
> ```
> 
> @ghalliday ?

hm, where is this failure, does't appear to be  any failed in the checks tab, was it rerun?

**Review by ghalliday — APPROVED** [review]
Looks sensible

File: system/jlib/jcontainerized.cpp [hunk 1]
```diff
@@ -255,7 +266,18 @@ void runJob(const char *componentName, const char *wuid, const char *jobName, co
...
-        deleteResource(componentName, "networkpolicy", jobName);
+    {
+        try
+        {
+            deleteResource(componentName, "networkpolicy", jobName);
+        }
+        catch(IException *e)
+        {
+            // we do not want this error to propagate to the workunit and cause it to fail, and mask other errors
+            OWARNLOG(e, "Failed to networkpolicy");
```
Comment (mckellyln) [inline system/jlib/jcontainerized.cpp:277]: Failed to **delete** networkpolicy ?

Comment (jakesmith) [reply system/jlib/jcontainerized.cpp:277]: I'll correct and squash in

**Review by mckellyln — APPROVED** [review]
Just small comment about msg.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**